platform/viewer/src/App.js

```diff
@@ -104,13 +110,30 @@ class App extends Component {
   constructor(props) {
     super(props);
 
-    this._appConfig = props;
+    this._appConfig = { props, ...props.config };
+
+    const { config, oidc, defaultExtensions } = props;
 
-    const { servers, extensions, hotkeys, oidc } = props;
+    const { servers, hotkeys, tools, extensions } =
+      typeof config === 'function'
+        ? config({
+            servicesManager,
+            dependencies: {
+              merge,
+              SimpleDialog,
+            },
+          })
+        : config;
 
     this.initUserManager(oidc);
-    _initServices([UINotificationService, UIModalService, UIDialogService]);
-    _initExtensions(extensions, hotkeys);
+    _initServices([
+      UINotificationService,
+      UIModalService,
+      UIDialogService,
+      UIContextMenuService,
+      UILabellingFlowService,
+    ]);
+    _initExtensions([...defaultExtensions, ...extensions], hotkeys, tools);
```

Comment:
@dannyrb I thought about adding them this way, what do you think? we can have the extensions coming from the config file and the defaultExtensions added using the index.js. Maybe more explicit? :mag_right: